lightning/src/offers/invoice_request.rs

```diff
@@ -588,6 +588,36 @@ pub struct InvoiceRequest {
 	signature: Signature,
 }
 
+impl InvoiceRequestContents {
+	pub(crate) fn amount_source(&self) -> Result<InvoiceRequestAmountSource, Bolt12SemanticError> {
```

Comment:
We should look into using `InvoiceRequestAmountSource` as the internal representation in `InvoiceRequestContents` to avoid needing a `Result`. Maybe similar for `Bolt12Invoice`.